### ISSUE TITLE
.Net 6 is end of support very soon. It's also required for the vcc cli.

### ISSUE THREAD

**Issue by theepicsnail** [vrchat-community/creator-companion#532]
.Net 6 is End of support November 12, 2024 (2 months from now).

https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core

VRChat Creator Companions CLI has a hard requirement on .Net 6

https://vcc.docs.vrchat.com/vpm/cli

Using the current version (.Net 8) results in an error saying .Net 6 is needed.

```
You must install or update .NET to run this application.

App: /home/username/.dotnet/tools/vpm
Architecture: x64
Framework: 'Microsoft.NETCore.App', version '6.0.0' (x64)
.NET location: /usr/lib/dotnet

The following frameworks were found:
  8.0.8 at [/usr/lib/dotnet/shared/Microsoft.NETCore.App]

Learn more:
https://aka.ms/dotnet/app-launch-failed

To install missing framework, download:
https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=6.0.0&arch=x64&rid=ubuntu.24.04-x64&os=linuxmint.22
```

Note for anyone finding this after it's removed from package managers, .Net is currently still downloadable through https://dotnet.microsoft.com/en-us/download/dotnet/6.0

**Comment by orels1** [comment]
Hey!
Thanks for reaching out. Yes, we're aware that support for .Net 6 is ending soon and we're going to be upgrading.

If we don't update before the end date - we'll update the docs with the download links for older versions.
But hopefully it won't come to that.

**Comment by orels1** [comment]
VCC and VCC CLI are moving to .NET 6 with VCC 2.4.0 (next major release)

**Comment by orels1** [comment]
There is a new pre-release of the CLI available that uses .NET 8

You can get it here https://www.nuget.org/packages/VRChat.VPM.CLI#versions-body-tab
or using the following command

```
dotnet tool install -g VRChat.VPM.CLI --version 0.1.28-beta.1
```

If you could try this out and confirm that this works correctly for you as well, that would be very appreciated!

**Comment by theepicsnail** [comment]
It works, thank you! 
Here's the list of steps I used to verify:

### Remove sdk-6.0 sdk and runtime
`sudo apt remove --purge dotnet-sdk-6.0`
`sudo apt remove --purge dotnet-runtime-6.0`

`dotnet --list-runtimes`

> Microsoft.AspNetCore.App 8.0.8 [/usr/lib/dotnet/shared/Microsoft.AspNetCore.App]
> Microsoft.NETCore.App 8.0.8 [/usr/lib/dotnet/shared/Microsoft.NETCore.App]

Note no dotnet 6. 


### Verify the current version no longer works in my environment by doing a fresh install and running:

`dotnet tool uninstall --global vrchat.vpm.cli`
> Tool 'vrchat.vpm.cli' (version '0.1.27') was successfully uninstalled.

`dotnet tool install --global vrchat.vpm.cli`

> Skipping NuGet package signature verification.
> You can invoke the tool using the following command: vpm
> Tool 'vrchat.vpm.cli' (version '0.1.27') was successfully installed.


`vpm`
> You must install or update .NET to run this application.
> 
> App: /home/snail/.dotnet/tools/vpm
> Architecture: x64
> Framework: 'Microsoft.NETCore.App', version '6.0.0' (x64)
> .NET location: /usr/lib/dotnet
> 
> The following frameworks were found:
>   8.0.8 at [/usr/lib/dotnet/shared/Microsoft.NETCore.App]
> 
> Learn more:
> https://aka.ms/dotnet/app-launch-failed
> 
> To install missing framework, download:
> https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=6.0.0&arch=x64&rid=ubuntu.22.04-x64&os=linuxmint.21.3
> 

### Switch to new version:
`dotnet tool uninstall --global vrchat.vpm.cli`
> Tool 'vrchat.vpm.cli' (version '0.1.27') was successfully uninstalled.

`dotnet tool install -g VRChat.VPM.CLI --version 0.1.28-beta.1`
> Skipping NuGet package signature verification.
> You can invoke the tool using the following command: vpm
> Tool 'vrchat.vpm.cli' (version '0.1.28-beta.1') was successfully installed.
> 

`vpm`
> [12:47:50 INF] Welcome to the CLI for the VRChat Package Manager. Use --help to find out what you can do.

**Comment by orels1** [comment]
Thank you for verifying!

**Comment by orels1** [comment]
This is now available fully in CLI 0.1.28 release
